test/intl402/DateTimeFormat/prototype/format/dayPeriod-long-en.js

```diff
@@ -6,88 +6,76 @@ esid: sec-createdatetimeformat
+// b) represented in sequence.
+var expectedDayPeriods = [
+  'in the morning',
+  'noon',
+  'in the afternoon',
+  'in the evening',
+  'at night'
+];
 
-const long = new Intl.DateTimeFormat('en', {
+// Cover all 24 hours of a single day.
+var inputs = [];
+for (var h = 0; h < 24; h++) {
+  inputs.push(new Date(2017, 11, 12,  h, 0, 0, 0));
+}
+
+// Verify complete and exclusive representation.
+var formatter = new Intl.DateTimeFormat('en', {
   dayPeriod: 'long'
 });
+var observedDayPeriods = [];
+var unexpectedDayPeriods = [];
+for (var h = 0; h < 24; h++) {
+  var dayPeriod = formatter.format(inputs[h]);
+  observedDayPeriods.push(dayPeriod);
+  if (expectedDayPeriods.indexOf(dayPeriod) === -1) {
+    unexpectedDayPeriods.push(dayPeriod);
+  }
+}
+var unusedDayPeriods = expectedDayPeriods.filter(function (dayPeriod) {
+  return observedDayPeriods.indexOf(dayPeriod) === -1;
+});
+assert.compareArray(unexpectedDayPeriods, [],
+  'unexpected dayPeriods: ' + unexpectedDayPeriods.join());
+assert.compareArray(unusedDayPeriods, [],
+  'unused dayPeriods: ' + unusedDayPeriods.join());
 
-assert.sameValue(long.format(d0000), 'at night', '00:00, long format');
-assert.sameValue(long.format(d0100), 'at night', '01:00, long format');
-assert.sameValue(long.format(d0200), 'at night', '02:00, long format');
-assert.sameValue(long.format(d0300), 'at night', '03:00, long format');
-assert.sameValue(long.format(d0400), 'at night', '04:00, long format');
-assert.sameValue(long.format(d0500), 'at night', '05:00, long format');
-assert.sameValue(long.format(d0600), 'in the morning', '06:00, long format');
-assert.sameValue(long.format(d0700), 'in the morning', '07:00, long format');
-assert.sameValue(long.format(d0800), 'in the morning', '08:00, long format');
-assert.sameValue(long.format(d0900), 'in the morning', '09:00, long format');
-assert.sameValue(long.format(d1000), 'in the morning', '10:00, long format');
-assert.sameValue(long.format(d1100), 'in the morning', '11:00, long format');
-assert.sameValue(long.format(d1200), 'noon', '12:00, long format');
-assert.sameValue(long.format(d1300), 'in the afternoon', '13:00, long format');
-assert.sameValue(long.format(d1400), 'in the afternoon', '14:00, long format');
-assert.sameValue(long.format(d1500), 'in the afternoon', '15:00, long format');
-assert.sameValue(long.format(d1600), 'in the afternoon', '16:00, long format');
-assert.sameValue(long.format(d1700), 'in the afternoon', '17:00, long format');
-assert.sameValue(long.format(d1800), 'in the evening', '18:00, long format');
-assert.sameValue(long.format(d1900), 'in the evening', '19:00, long format');
-assert.sameValue(long.format(d2000), 'in the evening', '20:00, long format');
-assert.sameValue(long.format(d2100), 'at night', '21:00, long format');
-assert.sameValue(long.format(d2200), 'at night', '22:00, long format');
-assert.sameValue(long.format(d2300), 'at night', '23:00, long format');
+function relativeIndex(arr, relIndex) {
```

Comment:
Suggestion: indicate that this is obviously a workaround.
```suggestion
function arrayAt(arr, relativeIndex) {
```
Or alternatively, implement it manually at the site of use:
```diff
-  var prevDayPeriod = relativeIndex(observedDayPeriods, h - 1);
+  var prevDayPeriod = observedDayPeriods[(h || observedDayPeriods.length) - 1];
   if (dayPeriod === prevDayPeriod) continue;
   transitionCount++;
   var i = expectedDayPeriods.indexOf(dayPeriod);
-  assert.sameValue(prevDayPeriod, relativeIndex(expectedDayPeriods, i - 1),
+  assert.sameValue(prevDayPeriod, expectedDayPeriods[(i || expectedDayPeriods.length) - 1],
     dayPeriod + ' must be preceded by ' + prevDayPeriod);
```

Reply:
I'll rename it `arrayAt`.